x-pack/legacy/plugins/reporting/server/legacy.ts

```diff
@@ -4,35 +4,75 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { Legacy } from 'kibana';
-import { PluginInitializerContext } from 'src/core/server';
+import { get } from 'lodash';
+import { take } from 'rxjs/operators';
+import { CoreSetup, PluginInitializerContext } from 'src/core/server';
+import { ConfigType, PluginsSetup } from '../../../../plugins/reporting/server';
 import { SecurityPluginSetup } from '../../../../plugins/security/server';
 import { ReportingPluginSpecOptions } from '../types';
 import { plugin } from './index';
-import { LegacySetup, ReportingStartDeps } from './types';
+import { LegacySetup, ReportingConfig, ReportingStartDeps } from './types';
 
 const buildLegacyDependencies = (
+  coreSetup: CoreSetup,
   server: Legacy.Server,
   reportingPlugin: ReportingPluginSpecOptions
-): LegacySetup => ({
-  config: server.config,
-  info: server.info,
-  route: server.route.bind(server),
-  plugins: {
-    elasticsearch: server.plugins.elasticsearch,
-    xpack_main: server.plugins.xpack_main,
-    reporting: reportingPlugin,
-  },
-});
+): LegacySetup => {
+  return {
+    route: server.route.bind(server),
+    plugins: {
+      xpack_main: server.plugins.xpack_main,
+      reporting: reportingPlugin,
+    },
+  };
+};
+
+const buildConfig = (
+  coreSetup: CoreSetup,
+  server: Legacy.Server,
+  reportingConfig: ConfigType
+): ReportingConfig => {
+  const config = server.config();
+  const { http } = coreSetup;
+  const serverInfo = http.getServerInfo();
+
+  const kbnConfig = {
+    path: {
+      data: config.get('path.data'), // FIXME: get from the real PluginInitializerContext
+    },
+    server: {
+      basePath: coreSetup.http.basePath.serverBasePath,
+      host: serverInfo.host,
+      name: serverInfo.name,
+      port: serverInfo.port,
+      uuid: coreSetup.uuid.getInstanceUuid(),
+      protocol: serverInfo.protocol,
+    },
+  };
+
+  // spreading arguments as an array allows the return type to be known by the compiler
```

Comment:
Very clever. Do you think (since we're using `get`) there's an opportunity for injection of any kind?

Reply:
The `reportingConfig` object would have to be polluted somehow. It doesn't look like that could happen in this part of the code though.